user-api/src/main/scala/com/example/auction/user/api/UserService.scala

```diff
@@ -3,35 +3,44 @@ package com.example.auction.user.api
 import java.util.UUID
 
 import akka.NotUsed
-import com.lightbend.lagom.scaladsl.api.{Service, ServiceCall}
+import com.lightbend.lagom.scaladsl.api.{Descriptor, Service, ServiceCall}
 import play.api.libs.json.{Format, Json}
-import com.example.auction.utils.JsonFormats._
 
 trait UserService extends Service {
-  def createUser: ServiceCall[CreateUser, User]
-  def getUser(userId: UUID): ServiceCall[NotUsed, User]
-
-  // Remove once we have a proper user service
   def getUsers: ServiceCall[NotUsed, Seq[User]]
+  def getUser(id: UUID): ServiceCall[NotUsed, User]
+  def createUser: ServiceCall[UserCreationRequest, User]
+  def updateUser(id: UUID): ServiceCall[UserUpdateRequest, User]
+  def deleteUser(id: UUID): ServiceCall[NotUsed, NotUsed]
 
-  def descriptor = {
+  override final def descriptor: Descriptor = {
     import Service._
     named("user").withCalls(
-      pathCall("/api/user", createUser),
+      pathCall("/api/user", getUsers _),
       pathCall("/api/user/:id", getUser _),
-      pathCall("/api/user", getUsers)
+      pathCall("/api/user", createUser _),
+      pathCall("/api/user/:id", updateUser _),
+      pathCall("/api/user/:id", deleteUser _)
     )
   }
 }
 
-case class User(id: UUID, name: String)
+case class User(id: UUID, username: String, email: String)
 
 object User {
-  implicit val format: Format[User] = Json.format
+  implicit val format: Format[User] = Json.format[User]
+}
+
+case class UserCreationRequest(username: String,
```

Comment:
Why use default values? If values can be blank maybe we shouls make the fields `Option[T]`